File: 叮当快药/get_sign.js
const  MD5 = require("crypto-js");

function get_sign():
    y = 'ddsy.product.query.orgcode.product.list.b2cmethodddsy.product.query.orgcode.product.list.b2corderTypeId0orgcode010101,010104otcMark1,2,99pageNo1pageSize100platH5platformH5shopId-1t2021-12-3 18:36:2v1.0versionName3.2.06C57AB91A1308E26B797F4CD382AC79D'
    v= MD5(y)
    e = "https://product.ddky.com/product/queryOrgcodeProductListForB2C.htm"
    l = [
        "method",
        "orderTypeId",
        "orgcode",
        "otcMark",
        "pageNo",
        "pageSize",
        "plat",
        "platform",
        "shopId",
        "t",
        "v",
        "versionName"
    ]

    for (var y, v , k = e + "?sign=" + v, w = 0; w < l.length; w++)
        (t.get(l[w]) + "").indexOf("+") >= 0 || -1 != (t.get(l[w]) + "").indexOf("&") ? k += "&" + l[w] + "=" + encodeURIComponent(t.get(l[w])) : "pageUrl" == l[w] ? k += "&" + l[w] + "=" + encodeURIComponent(t.get(l[w])) : k += "&" + l[w] + "=" + t.get(l[w]);
    return k

File: 叮当快药/get_sign.py
from hashlib import md5

def encrypt_md5(s):
    # 创建md5对象
    new_md5 = md5()
    # 这里必须用encode()函数对字符串进行编码，不然会报 TypeError: Unicode-objects must be encoded before hashing
    new_md5.update(s.encode(encoding='utf-8'))
    # 加密
    return new_md5.hexdigest()

orgcode = '011501,011502,011503,011504,011505,010301,010302,010303,010305,010306,010307,011601,011602,011603,011605,010201,010202,010203,011401,011402,011403,011404,010701,010702,010703,010704,010706,010707,010709,010710,010711,011101,011106'
page = '1'
t = '2022-01-10 16:08:52'
url_method = 'ddsy.product.query.orgcode.product.list.b2c'
url_ = f'method=ddsy.product.query.orgcode.product.list.b2c&orderTypeId=0&orgcode={orgcode}&otcMark=1,2,99&pageNo={page}&pageSize=100&plat=H5&platform=H5&shopId=-1&t={t}&v=1.0&versionName=3.2.0'

md5_str = url_method + url_

sign = encrypt_md5(md5_str)


url = f'https://product.ddky.com/product/queryOrgcodeProductListForB2C.htm?sign={sign}&{url_}'

print(url)
